cirq-core/cirq/value/measurement_key.py

```diff
@@ -104,16 +104,26 @@ def parse_serialized(cls, key_str: str):
     def _with_key_path_(self, path: Tuple[str, ...]):
         return self.replace(path=path)
 
-    def _with_key_path_prefix_(self, prefix: Tuple[str, ...]):
-        return self._with_key_path_(path=prefix + self.path)
+    def _with_key_path_prefix_(
+        self,
+        path: Tuple[str, ...],
+        local_keys: FrozenSet['MeasurementKey'],
+        extern_keys: FrozenSet['MeasurementKey'],
+    ):
+        if self in local_keys:
+            raise ValueError(f'Conflicting measurement keys found: {self}')
+        new_key = self.replace(path=path + self.path)
+        if new_key in extern_keys:
+            raise ValueError(f'Conflicting measurement keys found: {new_key}')
```

Comment:
It might be worthwhile to give this a distinct error message from the local collision, since external collisions are likely more difficult to spot.

Reply:
I originally had that, but I actually found the error message unintuitive. Ultimately it'll report the fully scoped key e.g. '0:1:A', so that should be enough information for the user to find them. Saying "external" or "internal" in the error message just made it confusing IMO.